### DISCUSSION TITLE
[3.x] How to render nested blocks from settings block?

### DISCUSSION THREAD

**Discussion by tijsbakker** [area17/twill#1945] [Q&A] [answered]
Hi @haringsrob, 

I have a settings section called `footer.info` containing a block editor called `downloads`. This all works fine in the CMS, but I can't seem to render the blocks from the editor within a blade template in any way.

The settings documentation states: `Get the full block model for the settings. This is especially useful if you have a block editor in the settings and you need to render those.`

From the nested-block documentation I get that now I should do something like `$renderData->renderChildren('downloads')`. The `$renderData` isn't available within my blade component so I should do something like:

`$this->info = \TwillAppSettings::getGroupDataForSectionAndName('footer','info');`

`$renderData = new \A17\Twill\Services\Blocks\RenderData($this->info, 'info', $this->info->children->toArray());`

To which I receive an exception: Attempt to read property "renderData" on array. 

I don't get this at all, please help and thanks in advance!

Tijs

**Comment by haringsrob** [comment, marked as answer]
Hey @tijsbakker 

Sorry for the late reply, but this was not yet properly implemented. 

I added the feature in https://github.com/area17/twill/pull/1988

**Reply by tijsbakker** [reply]
Hi @haringsrob,

Thanks!
